docs/DeveloperGuide.md

```diff
@@ -2008,142 +2030,296 @@ testers are expected to do more *exploratory* testing.
 
-`add n/Thomas s/Bedok p/12345678 e/thomas@gmail.com t/@thomas`
+#### **_Scenario 1_**
 
-Expected Output in the Person List: New person added to EduMate.
+Context: There is no contact named "Thomas" in `EduMate`
 
-`add n/Edward`
+Command executed: `add n/Thomas s/Bedok p/12345678 e/thomas@gmail.com t/@thomas`
 
-Expected Output in the Command Output Box: Error message for invalid command format.
+Expected Output in the Person List: New person added to `EduMate`.
+
+#### **_Scenario 2_**
+
+Context: There is contact named "Edward" in `EduMate`
+
+Command executed: `add n/Edward`
+
+Expected Output in the Command Response Box: `This person already exists in the address book`
 
 ### **View a person**
-`view`
 
-Expected Output in the Profile Panel: The user's (you) profile is shown.
+#### **_Scenario 1_**
+
+Context: None needed.
+
+Command executed: `view`
 
-`view 5`
+Expected Output in the Profile Panel: The user's profile is shown.
 
-Expected Output in the Profile Panel: The fifth person's profile is shown.
+#### **_Scenario 2_**
 
-`view n/Thomas`
+Context: There exists a `Person` with `ContactIndex 5` in `EduMate`.
+
+Command executed: `view 5`
+
+Expected Output in the Profile Panel: The `Person` with `ContactIndex 5` is shown.
+
+#### **_Scenario 3_**
+
+Context: Thomas' profile is present in `EduMate`.
+
+Command executed: `view n/Thomas`
 
 Expected Output in the Profile Panel: Thomas' profile is shown.
 
 ### **Edit a person**
 
-`edit 3 n/Henry`
+#### **_Scenario 1_**
+
+Context: There exists a `Person` with `ContactIndex 3` in `EduMate`.
+
+Command executed: `edit 3 n/Henry`
 
-Expected Output in the Person List: The third person's name is changed to Henry.
+Expected Output in the Person List: The Person Card with `ContactIndex 3` contains the name Henry.
 
-`edit n/Gordon`
+#### **_Scenario 2_**
+
+Context: None needed.
+
+Command executed: `edit n/Gordon`
 
 Expected Output in Profile Panel: The user's name is changed to Gordon.
 
 ### **Delete a person**
 
-`delete 4`
+#### **_Scenario 1_**
+
+Context: There exists a `Person` with `ContactIndex 4` in `EduMate`.
+
+Command executed: `delete 4`
+
+Expected Output in the Person List: Person Card containing profile of `Person` with `ContactIndex 4` is removed.
+
+#### **_Scenario 2_**
 
-Expected Output in the Person List: The fourth person has been removed, and there is no fourth index.
+Context: There exists a `Person` with `ContactIndex 1, 2, 3` and `4` in `EduMate` and there is no other contact with the name "James".
 
-`delete 4`, `add n/James e/james@gmail.com t/@james s/Bishan p/87654321`
+Command executed: `delete 4`, `add n/James e/james@gmail.com t/@james s/Bishan p/87654321`
 
-Expected Output in the Person List: New person has been added to EduMate, with an index of 4.
+Expected Output in the Person List: Person Card containing `Person` with `ContactIndex 4` is removed. A new `Person` named James is created and assigned `ContactIndex 4` and appears on Person List.
```

Comment:
```suggestion
Expected Output in the PersonList: Person Card containing `Person` with `ContactIndex 4` is removed. A new `Person` named James is created and assigned `ContactIndex 4` and appears on Person List.
```

Remove the space in Person List, since the following sections don't have the space.